test/Microsoft.Extensions.Primitives.Tests/StringValuesTests.cs

```diff
@@ -302,5 +333,81 @@ public void Concat(StringValues stringValues, string[] array)
             string[] expectedAppended = filled.Concat(array).ToArray();
             Assert.Equal(expectedAppended, StringValues.Concat(new StringValues(filled), stringValues));
         }
+
+        [Theory]
+        [MemberData(nameof(FilledStringValuesWithExpectedStrings))]
+        public void StringValuesEquals_String(StringValues stringValues, string expected)
+        {
+            var notEqual = "bcd";
+
+            Assert.True(StringValues.Equals(stringValues, expected));
+            Assert.True(StringValues.Equals(expected, stringValues));
+            Assert.True(StringValues.Equals(stringValues, new StringValues(expected)));
+            Assert.True(StringValues.Equals(new StringValues(expected), stringValues));
+
+            Assert.True(StringValues.Equals(stringValues.ToArray(), new StringValues(expected)));
+            Assert.True(StringValues.Equals(new StringValues(expected), stringValues.ToArray()));
+            Assert.True(StringValues.Equals(new StringValues(stringValues.ToArray()), new StringValues(expected)));
+            Assert.True(StringValues.Equals(new StringValues(expected), new StringValues(stringValues.ToArray())));
+
+            // StringValues.ToString converts null to "" so skip for null values
+            if (expected != null)
+            {
+                Assert.True(StringValues.Equals(stringValues.ToString(), new StringValues(expected)));
+                Assert.True(StringValues.Equals(new StringValues(expected), stringValues.ToString()));
+                Assert.True(StringValues.Equals(new StringValues(stringValues.ToString()), new StringValues(expected)));
+                Assert.True(StringValues.Equals(new StringValues(expected), new StringValues(stringValues.ToString())));
+            }
+
+            Assert.False(StringValues.Equals(stringValues, notEqual));
+            Assert.False(StringValues.Equals(notEqual, stringValues));
+            Assert.False(StringValues.Equals(stringValues, new StringValues(notEqual)));
+            Assert.False(StringValues.Equals(new StringValues(notEqual), stringValues));
+
+            Assert.False(StringValues.Equals(stringValues.ToArray(), new StringValues(notEqual)));
+            Assert.False(StringValues.Equals(new StringValues(notEqual), stringValues.ToArray()));
+            Assert.False(StringValues.Equals(new StringValues(stringValues.ToArray()), new StringValues(notEqual)));
+            Assert.False(StringValues.Equals(new StringValues(notEqual), new StringValues(stringValues.ToArray())));
+
+            Assert.False(StringValues.Equals(stringValues.ToString(), new StringValues(notEqual)));
+            Assert.False(StringValues.Equals(new StringValues(notEqual), stringValues.ToString()));
+            Assert.False(StringValues.Equals(new StringValues(stringValues.ToString()), new StringValues(notEqual)));
+            Assert.False(StringValues.Equals(new StringValues(notEqual), new StringValues(stringValues.ToString())));
```

Comment:
Removed a bunch of the tests, but also added a few more e.g. for `operator==`, `object` and instance that weren't being tested.